scripts/e2e-full/phases/90-cleanup.sh

```
@@ -54,6 +54,20 @@ run_phase() {
     info "killed metro pid $E2E_METRO_PID"
   fi
 
+  # Reap the fake provider from the device-provider tier: its own
+  # simulator-server and its descriptor. Argent deliberately never deletes
+  # another process's descriptor, so the harness — which IS that process here —
+  # has to.
+  for pid in "${E2E_PROVIDER_SIM_PID:-}"; do
```

Shellcheck (0.11.0)

[error] Since you double quoted this, it will not word split, and the loop will only run once.

(SC2066)